cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/WidgetAttributeInput/AttributeMultipleValuesInput/index.tsx

```diff
@@ -0,0 +1,319 @@
+ * For instance, users will be able to set following widget attributes.
+ *
+ * Example 1)
+ *     "options": [
+ *       {
+ *         "id": "true",
+ *        "label": "true"
+ *       },
+ *       {
+ *         "id": "false",
+ *         "label": "false"
+ *       }
+ *     ]
+ * Example 2)
+ *     "values": [
+ *       "json",
+ *       "xml",
+ *       "tsv",
+ *       "csv",
+ *       "text",
+ *       "blob"
+ *     ],
+ *
+ * As we can see from the examples above, this can be an array of string or an array of object (ID-Label pair).
+ * In this component, a prop named 'supportedTypes' stores all the available types for a certain field.
+ *
+ * For instance, 'select' widget has an attribute field called 'options'.
+ * 'options' can be of different types, therefore its 'supportedTypes' is as following:
+ *     ['ISelectOptions[]', 'string[]', 'number[]']
+ * If the user selects ISelectOptions[], then it that means the 'options' field will be
+ * an array of ISelectOptions.
+ * Since ISelectOptions interface is defined to be a 'value' and 'label' pair,
+ * our component should render multiple input rows of key-value pairs.
+ *
+ * Some types such as 'IDropdownOption', can be string, number, or value-label pair.
+ *     export type IDropdownOption = string | number | IComplexDropdown;
+ *     interface IComplexDropdown {
+ *       value: string | number;
+ *       label: string;
+ *     }
+ * Therefore, we should futher process IDropdownOption[], by adding all the available types in the function
+ * 'processSupportedTypes'.
+ *     supportedTypes.add(SupportedType.String);
+ *     supportedTypes.add(SupportedType.Number);
+ *     supportedTypes.add(SupportedType.ValueLabelPair);
+ */
+
+export enum SupportedType {
+  String = 'string',
+  Number = 'number',
+  ValueLabelPair = 'Value-Label pair',
+  IDLabelPair = 'ID-Label pair',
+}
+
+const styles = (): StyleRules => {
+  return {
+    typeSelectInput: {
+      marginTop: '5px',
+      marginBottom: '5px',
+      width: '20%',
+    },
+  };
+};
+
+export const processSupportedTypes = (types: string[]) => {
+  const allTypes = new Set();
+  types.forEach((type) => {
+    switch (type) {
+      case 'string[]':
+        allTypes.add(SupportedType.String);
+        break;
+      case 'number[]':
+        allTypes.add(SupportedType.Number);
+        break;
+      case 'IOption[]':
+        allTypes.add(SupportedType.IDLabelPair);
+        break;
+      case 'IDropdownOption[]':
+        allTypes.add(SupportedType.String);
+        allTypes.add(SupportedType.Number);
+        allTypes.add(SupportedType.ValueLabelPair);
+        break;
+      case 'ISelectOptions[]':
+        allTypes.add(SupportedType.ValueLabelPair);
+        break;
+      case 'FilterOption[]':
+        allTypes.add(SupportedType.String);
+        allTypes.add(SupportedType.IDLabelPair);
+        break;
+      default:
+        break;
```

Comment:
if it's not supported i think just display a textbox

Reply:
Resolved.

Adding `SupportedType.String` in case the type is not supported.